### PR TITLE
Jar files for everyone

### PR DESCRIPTION
**Pull request by klaren**
This fix jar file generation together with javadoc generation

There are still issues with `meta packages` and standalone builds. This will be addressed next.
Clirr settings is not updated either but since it;s scheduled to be replaced by revapi this is not a priority.

### CODE REVIEW

File: community/neo4j/pom.xml [hunk 1]
```diff
@@ -17,11 +17,9 @@
   <description>A meta package containing the most used Neo4j libraries. Intended use: as a Maven dependency.</description>
 
   <properties>
-    <short-name>neo4j</short-name>
-    <version-package>impl.neo4j</version-package>
     <license-text.header>GPL-3-header.txt</license-text.header>
-    <javadoc.skip>false</javadoc.skip>
     <licensing.prepend.text>notice-gpl-prefix.txt</licensing.prepend.text>
+    <maven.javadoc.skip>true</maven.javadoc.skip> <!-- TODO: fix -->
```
Comment (MishaDemianenko) [inline community/neo4j/pom.xml:22]: todo

File: community/server/pom.xml [hunk 2]
```diff
@@ -87,6 +84,183 @@
...
+                <descriptor>src/main/assemblies/test-sources-assembly.xml</descriptor>
+              </descriptors>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.neo4j.build.plugins</groupId>
+        <artifactId>license-maven-plugin</artifactId>
```
Comment (MishaDemianenko) [inline community/server/pom.xml:150]: hm do we need this module to have custom license plugin config?

File: community/server/pom.xml [hunk 3]
```diff
@@ -87,6 +84,183 @@
...
+          <execution>
+            <id>attach-docs</id>
+            <phase>none</phase>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
```
Comment (MishaDemianenko) [inline community/server/pom.xml:194]: and surefire?

File: community/server/pom.xml [hunk 4]
```diff
@@ -87,6 +84,183 @@
...
+        <configuration>
+          <systemPropertyVariables>
+            <java.io.tmpdir>${project.build.directory}</java.io.tmpdir>
+          </systemPropertyVariables>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
```
Comment (MishaDemianenko) [inline community/server/pom.xml:204]: not sure why do we need those profiles at all here?

File: community/server/pom.xml [hunk 5]
```diff
@@ -87,6 +84,183 @@
...
+        <repository>
+          <id>neo4j-dev</id>
+          <name>Neo4j Developer Repository</name>
+          <url>http://m2.neo4j.org/content/groups/everything/</url>
+        </repository>
+      </repositories>
+    </profile>
+  </profiles>
+
+  <repositories>
```
Comment (MishaDemianenko) [inline community/server/pom.xml:246]: and separate repos

File: enterprise/neo4j-enterprise/pom.xml [hunk 6]
```diff
@@ -18,154 +18,15 @@
...
-    <dependency>
-      <groupId>org.neo4j</groupId>
-      <artifactId>neo4j-common</artifactId>
-      <version>${project.version}</version>
-      <type>test-jar</type>
-      <scope>test</scope>
-    </dependency>
-  </dependencies>
+  <properties>
+    <maven.javadoc.skip>true</maven.javadoc.skip> <!-- TODO: fix -->
```
Comment (MishaDemianenko) [inline enterprise/neo4j-enterprise/pom.xml:28]: todo

File: pom.xml [hunk 7]
```diff
@@ -151,131 +151,151 @@
...
+              <group>
+                <title>Server Plugin API</title>
+                <packages>org.neo4j.server:org.neo4j.server.*</packages>
+              </group>
+              <group>
+                <title>Helpers</title>
+                <packages>org.neo4j.helpers:org.neo4j.helpers.*</packages>
+              </group>
+              <group>
+                <title>Legacy Indexes</title>
```
Comment (MishaDemianenko) [inline pom.xml:233]: Not a legacy anymore - <i>Implicit</i>

**Review by MishaDemianenko — COMMENTED** [review]
Couple of minor comments

File: pom.xml (hunk 7 above)
Comment (klaren) [reply pom.xml:233]: Explicit?

Comment (MishaDemianenko) [reply pom.xml:233]: ah yes 👍  naming things.....

**Pull request merged**